lib/Paws/Net/RetryCallerRole.pm

```diff
@@ -15,8 +15,7 @@ package Paws::Net::RetryCallerRole;
     do {
       $tracker->one_more_try;
 
-      my ($status, $content, $headers) = $self->send_request($service, $call_object);
-      my $result = $self->caller_to_response($service, $call_object, $status, $content, $headers);
+      my $result = $self->send_request($service, $call_object);
```

Comment:
pplu:feat/s3_support currently has the following two lines in lib/Paws/Net/RetryCallerRole.pm

<pre>
      my ($status, $content, $headers) = $self->send_request($service, $call_object);
      my $result = $self->caller_to_response($service, $call_object, $status, $content, $headers);
</pre>

This causes a problem because $self->send_request calls $self->caller_to_response internally and returns the response result. $self->caller_to_response is then called again with the response result as arguments and the whole thing breaks when there is an error constructing the response and a Paws::Exception is returned as an argument to the `caller_to_response` method